### PR TITLE
machine/usb: allow USB Endpoint settings to be changed externally

### PR DESCRIPTION
**Pull request by sago35**
This PR is configured from machine/usb/xxx instead of machine using the changes in #3849.
This PR needs to be merged after #3849.

### CODE REVIEW

**Comment by deadprogram** [comment]
You can rebase this PR now @sago35 thank you!

**Comment by sago35** [comment]
Rebased. @deadprogram

File: src/machine/usb/config.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package usb
+
+type EndpointConfig struct {
+	No        uint8
```
Comment (deadprogram) [inline src/machine/usb/config.go]: This might be better named `Num` or perhaps `ID`?

Comment (sago35) [reply src/machine/usb/config.go]: Perhaps `Index` would be a better name.

Comment (sago35) [reply src/machine/usb/config.go]: Renamed.

**Comment by deadprogram** [comment]
Thank you for the improvements @sago35 now merging.

**Pull request merged**